ui/apps/platform/src/routePaths.js

```diff
@@ -99,6 +99,9 @@ export const vulnerabilitiesWorkloadCveImageSinglePath = `${vulnerabilitiesBaseP
 export const vulnerabilitiesWorkloadCveDeploymentSinglePath = `${vulnerabilitiesBasePath}/workload-cves/deployments/:deploymentId`;
 
 export const vulnerabilityReportingPath = `${vulnerabilitiesBasePath}/vulnerability-reporting`;
+export const vulnerabilityReportingCreatePath = `${vulnerabilityReportingPath}/create`;
+export const vulnerabilityReportingEditPath = `${vulnerabilityReportingPath}/edit/:reportId`;
+export const vulnerabilityReportingClonePath = `${vulnerabilityReportingPath}/clone/:reportId`;
```

Comment:
I wanted some input on the structure of these pages. The way I have them structured is as follows:

```
Reports list - /vulnerability-reporting
Create report - /vulnerability-reporting/create
Edit report - /vulnerability-reporting/edit/:reportId
Clone report - /vulnerability-reporting/clone:/reportId
```

I know we have another approach in other features that would look something like this:

```
Reports list - /vulnerability-reporting
Create report - /vulnerability-reporting?action=create
Edit report - /vulnerability-reporting/:reportId?action=edit
Clone report - /vulnerability-reporting/:reportId?action=clone
```

I do like the first approach more, especially in this situation. Some pros include:
- Clean and easy-to-read URL structure
- It works well because of the clear and distinct pages that correspond to each route
- There's a direct map between the URL and component

I can also see the pros of the second approach:
- More flexibility in state manipulation without changing the route
- Reduces the number of routes we need to manage

However, I don't think this particular feature requires anything complex in regard to manipulating the states. The URL also wouldn't look as clean and readable. So based on these points, I find that the approach I took for this feature would probably be the better option. Let me know if you think differently and/or your thoughts on the above

(cc: @vjwilson @pedrottimark @dvail @alwayshooin @bradr5 )

Reply:
I don't feel super strongly either way - the approach you took here looks good and it does make the URL cleaner. 

The one nit I have is that I would put `:reportId` before the action. To me it reads better if the URL is a hierarchical path to an entity, e.g.:

`/vulnerability-reporting` -> Get me the page for vuln reports
`/vulnerability-reporting/:reportId` -> Get me the page for _this_ vuln report
`/vulnerability-reporting/:reportId/edit` -> Get me _this_ vuln report's _edit_ page

vs 

`/vulnerability-reporting` -> Get me the page for vuln reports
`/vulnerability-reporting/edit` -> Get me the edit page for vuln reports (what entity is this without a report id?)
`/vulnerability-reporting/edit/:reportId` -> Get me the edit page loaded with _this_ report's details

It also makes related actions for a single entity further apart `/edit/:reportId` and `/create/:reportId` move actions on entity `:reportId` under two separate umbrellas, while `/:reportId/edit` and `/:reportId/create` are actions under a single entity's umbrella.

I think there might also be some disconnect for an implementation of a read only view, where the action disappears from the middle of the URL:
`/vulnerability-reporting/edit/:reportId` -> Edit
`/vulnerability-reporting/:reportId` -> View

(Unless `/view/` was an action that was explicit in the URL as well.)

Reply:
Those are some good points @vjwilson @dvail. I agree that the "view report details" route would then need to follow the same pattern. So we would have to do something like:

```
/vulnerability-reporting/view/:reportId
```

In terms of having the `:reportId` showing up before or after the noun, I have a few thoughts on that. I do think having it before makes it more predictable and readable. As Van mentioned, `there is still the ambiguity of sometimes URL part after the section is a noun (reportId) and sometimes it a verb (like "create")`. So, with the above addition, this would make it consistent. Also, I do think it's better in case we need further nested resources. For example, something like:

```
/vulnerability-reporting/view/:reportId/run-history
```

![Screenshot 2023-06-22 at 10 01 43 AM](https://github.com/stackrox/stackrox/assets/4805485/524d382b-9bb8-4fbf-95a1-10f1e088b424)

If we need to add this view as part of the URL, we could do that using this method. If we had the `:reportId` before, then it probably wouldn't read well.

But, I also agree that it would be less RESTful since usually, the resource comes right after the base path

**tldr;**

**Pros**:
- Predictable and readable
- Better for nested resources

**Cons**:
- Less RESTful

Reply:
Also, I do notice that when writing code for the `?action=x` approach, it does require more work in regards to parsing the query params and adding conditionals in the code to determine what component to render. Seems a little odd to me to do that ourselves when react-router could do that using the route switch component and pre-determined routes.

Reply:
Would be interested to hear the rest of the team's opinions on this as well (cc: @pedrottimark @alwayshooin @bradr5)

Reply:
I don't have a strong preference either way so I'd lean towards consistency. The first approach (`/:reportId/edit`) seems slightly more intuitive, but at the cost of flexibility. @sachaudh you do bring up a good point about having to parse the "action". That does sound a little odd to me, but probably not uncommon.

Reply:
When REST is so widespread for API, it seems like extra mental effort to swim against the current:
* endpoint address represents **noun** so page address can too
* request method represents **verb** so query parameter seems like closest available analogy

If I understand the discussion in this thread:
* noun verb id: more convenient to implement but less semantic
* noun id verb: more semantic but less convenient to implement

Is it a correct impression that the amount of less convenient is similar between verb after noun and id:
* in page address
* in query parameter

Side comment comparing 1.0 to 2.0 and leaving aside representation of verb:
* /main/vulnerability-management/reports/4bce65b5-f4ac-4548-9bfd-b991f8e3ea89
* /main/vulnerability-reports/id seems more noun-ish than /main/vulnerability-reporting/id

Reply:
@pedrottimark that last side comment did bring up an issue I didn't see before. Right now we have this as the route for viewing a single report (regardless of what we're deciding to do with regards to the route management):

```
/main/vulnerabilities/vulnerability-reporting/:reportId
```

I agree it could be improved by either doing something like

```
/main/vulnerabilities/vulnerability-reporting/reports/:reported

or

/main/vulnerabilities/reports/:reportId
```

The latter probably being simpler and shorter.

Reply:
Given the comments and suggestions from everyone, this is what I'm seeing:
- Van: In stronger support of query params method
- Dave: No strong opinion either way but suggests adjustments if doing the URL path method
- Brad: No strong opinion either way
- Mark: In stronger support of query params method and also some adjustments with the base path

Given everyone's comments and opinions, I think I'll go ahead and try to implement this using the query params method. Seems like even with the Workload CVEs feature, the query params are used to keep track of the tabs. So let's keep things consistent and try doing it this way. Thanks folks 👍🏼